torchprime/sharding/shard_model.py

```diff
@@ -197,7 +197,7 @@ def shard_param(tensor, spec: tuple[str, ...]):
     # and models are usually constructed eagerly in torchax.
     return torch_view(
       jax.make_array_from_callback(
-        tensor.shape, sharding, lambda slice_index: tensor[slice_index]
+        tensor.shape, sharding, lambda slice_index: jax_view(tensor[slice_index])
```

Comment:
for my understanding, why is this needed, and is there an impact on performance?

Reply:
Ah, this was just a bug. It probably used to work accidentally.

This code shards the model weights for torchax models. It does so by indexing into an unsharded torch tensor and giving views into that tensor to the `jax.make_array_from_callback` function.

The `jax.make_array_from_callback` function only works with JAX arrays. So the right thing to do is to use `jax_view` to wrap the torch tensor into a jax array before giving it to jax. It looks like the newer JAX version that comes from the updated PyTorch/XLA pin has enforced stricter checks that surfaced this bug.

In any case, none of this applies to `torch_xla` models and I had to fix the `torchax` specific tests as part of updating to PyTorch/XLA Jun 17.